frontend/src/components/SearchModal.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { throttle } from 'lodash'
+
+import Modal from './Modal'
+import { RecipeItem as Recipe } from './RecipeItem'
+import { searchRecipes } from '../store/actions'
+import { classNames } from '../classnames'
+
+const SEARCH_THROTTLE_MS = 100
+
+class SearchModal extends React.Component {
+
+  constructor (props) {
+    super(props)
+    this.inputRef = React.createRef()
+    this.search = throttle(this.props.search, SEARCH_THROTTLE_MS)
+  }
+
+  state = {
+    query: '',
+    show: false,
+  }
+
+  handleInputChange = (event) => {
+    this.setState({ query: event.target.value })
+  }
+
+  handleSearch = (event) => {
+    if (event.key === 'Enter') {
+      this.search(this.state.query)
+    }
+  }
+
+  handleKeyPress = (event) => {
+    if (document.activeElement.tagName !== 'BODY') return
+    const pressF = event.key === 'f' && !event.ctrlKey && !event.meta
```

Comment:
What about shift?

Reply:
that's covered by `event.key`. If we wanted `shift-f`, we'd check for `F`, not `f`